applications/GeoMechanicsApplication/custom_elements/transient_Pw_line_element.h

```diff
@@ -0,0 +1,419 @@
+                + r_properties[POROSITY] / r_properties[BULK_MODULUS_FLUID];
+        } 
+        else {
+            result = (biot_coefficient - r_properties[POROSITY]) / r_properties[BULK_MODULUS_SOLID] 
+                + r_properties[POROSITY] / TINY;
+        }
+
+        RetentionLaw::Parameters RetentionParameters(GetProperties(), rCurrentProcessInfo);
+        double degree_of_saturation = mRetentionLawVector[integration_point_index]->CalculateSaturation(RetentionParameters);
+        double derivative_of_saturation = mRetentionLawVector[integration_point_index]->CalculateDerivativeOfSaturation(RetentionParameters);
+
+        result *= degree_of_saturation;
+        result -= derivative_of_saturation * r_properties[POROSITY];
+        return result;
+    }
+
+
+    void InitializeSolutionStep(const ProcessInfo& rCurrentProcessInfo) override
+    {
+        RetentionLaw::Parameters RetentionParameters(this->GetProperties(), rCurrentProcessInfo);
+        for (unsigned int integration_point_index = 0;
+             integration_point_index < GetGeometry().IntegrationPointsNumber(GetIntegrationMethod());
+             ++integration_point_index) {
+            mRetentionLawVector[integration_point_index]->InitializeSolutionStep(RetentionParameters);
+        }
+    }
+    
+    
+    void FinalizeSolutionStep(const ProcessInfo& rCurrentProcessInfo) override
+    {
+        RetentionLaw::Parameters RetentionParameters(this->GetProperties(), rCurrentProcessInfo);
+        for (unsigned int integration_point_index = 0;
+             integration_point_index < GetGeometry().IntegrationPointsNumber(GetIntegrationMethod());
+             ++integration_point_index) {
+            mRetentionLawVector[integration_point_index]->FinalizeSolutionStep(RetentionParameters);
+        }
+    }
+
+
+    array_1d<double, TNumNodes> CalculateFluidBodyVector(
+        const Matrix& r_N_container,
+        const GeometryType::ShapeFunctionsGradientsType& dN_dX_container,
+        const ProcessInfo& rCurrentProcessInfo,
+        const Vector& rIntegrationCoefficients) const
+    {
+        std::size_t number_integration_points = GetGeometry().IntegrationPointsNumber(GetIntegrationMethod());
+        GeometryType::JacobiansType J_container;
+        J_container.resize(number_integration_points, false);
+        for (std::size_t i = 0; i < number_integration_points; ++i) {
+            J_container[i].resize(GetGeometry().WorkingSpaceDimension(), GetGeometry().LocalSpaceDimension(), false);
+        }
+        GetGeometry().Jacobian(J_container, this->GetIntegrationMethod());
+
+        const auto& r_properties = GetProperties();
+        BoundedMatrix<double, 1, 1> constitutive_matrix;
+        GeoElementUtilities::FillPermeabilityMatrix(constitutive_matrix, r_properties);
+
+        RetentionLaw::Parameters    RetentionParameters(GetProperties(), rCurrentProcessInfo);
+
+        array_1d<double, TNumNodes * TDim> VolumeAcceleration;
```

Comment:
Sorry the local variables shall bein snake_case style, like volume_acceleration and body_acceleration and others below.